maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/SurefireHelper.java

```diff
@@ -154,7 +155,11 @@ public static void reportExecution( SurefireReportParameters reportParameters, R
             return;
         }
 
-        if ( reportParameters.isTestFailureIgnore() )
+        if ( firstForkException instanceof SurefireBooterForkException )
+        {
+            throwException( reportParameters, result, firstForkException );
```

Comment:
throwException() has two behaviors. You do not have possibility to switch between them. Here must be only one behavior, means `BUILD ERROR` and not `BUILD FAILURE`. I know that you saw my code proposal but you should think about it. This situation is the same situation with the existing situation  `isFatal()` that we use nearby in this class. I think you want to have this change because it is an exception in the behavior of `-Dmaven.test.failure.ignore`.

Reply:
I feel that it is not important if build stop with error or failure ... simply should be break.

Reply:
both exception generate `BUILD FAILURE` at the end (probably a maven core bug)
but anyway we need a build stop no need of a distinction really.

Reply:
no need of distinction between both exception maven core doesn't handle for very long time https://github.com/apache/maven/pull/632
this mean we can simplify the surefire code even more

Reply:
@olamy 
@slawekjaranowski 
It is extremely important due to you want to interrupt the build with **ignoring the failure** but on the other side you let the build to say **BUILD FAILURE**. How come? You want to ignore it. Hopefully, we have another alternative **BUILD ERROR** which looks much more reasonable to the users and everyone I guess.

Here matters what you do and how you do it. This is 50% of the behavior of `maven.test.failure.ignore` because yes you stopped the build, but the next 50% is how you present the information to the client.

I will never use the same terminology in the text with "LOL" same way as Olivier openly does later in these comments, because this is disgusting. So what are we going to play, a game with ego, or we would patiently evaluate technical/business arguments?
@hboutemy 
Herve, pls participate as an independent moderator here. I expect from @olamy to be neutral and fully rational.

Reply:
it seems people don't agree on what `BUILD_FAILURE` and `BUILD_ERROR` mean or have as concrete result (which I personally don't know)
and I don't know why, instead of clarifying, the discussion becomes heated for no reason

Reply:
sample here https://github.com/olamy/maven-exception-plugin

so there is no difference between `throw MojoExecutionException` and `throw MojoFailureException`
we can definitely simplify the code managing this but should be part of another PR